assets/src/components/SidebarFields/ImageSidebarField.js

```diff
@@ -1,35 +1,36 @@
-import { MediaUpload, MediaUploadCheck } from '@wordpress/block-editor';
-import { Button } from '@wordpress/components';
-import { ImageHoverControls } from '../ImageHoverControls';
-import { __ } from '@wordpress/i18n';
+import {MediaUpload, MediaUploadCheck} from '@wordpress/block-editor';
+import {Button} from '@wordpress/components';
+import {ImageHoverControls} from '../ImageHoverControls';
 
-export const ImageSidebarField = ({ value, setValue, label }) => (
+const {__} = wp.i18n;
+
+export const ImageSidebarField = ({value, setValue, label}) => (
   <div className="components-base-control mb-3">
-    <label className="components-base-control__label mb-2 d-block">
+    <label className="components-base-control__label mb-2 d-block" htmlFor={value.id}>
       {label}
     </label>
     <MediaUploadCheck>
       <MediaUpload
+        id={value.id}
```

Comment:
It looks like the MediaUpload component [doesn't accept `id` as a prop](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/media-upload/README.md), so I'm not sure how that works 🤔

Reply:
Maybe the id should instead be set inside of the `render`